pkg/controller/catalog/storage/mem_storage_test.go

```diff
@@ -20,278 +20,242 @@ import (
 	"strings"
 	"testing"
 
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+	kapi "k8s.io/kubernetes/pkg/api"
 )
 
 const (
-	brokerOneUUID = "126b8154-a24a-4e79-9185-3df2eb4d18a8"
-	brokerTwoUUID = "2b0c42ed-c43a-4724-b883-e5ba878a8bfd"
+	brokerName1 = "Test1"
 )
 
 func TestNoBrokers(t *testing.T) {
-	s := CreateMemStorage()
-	l, err := s.ListBrokers()
+	const bogusBrokerName = "NOT THERE"
+	bs := CreateMemStorage().Brokers()
+	l, err := bs.List()
 	if err != nil {
-		t.Fatalf("ListBrokers failed with: %#v", err)
+		t.Fatalf("List failed with: %s", err)
 	}
 	if len(l) != 0 {
 		t.Fatalf("Expected 0 brokers, got %d", len(l))
 	}
-	b, err := s.GetBroker("NOT THERE")
+	b, err := bs.Get(bogusBrokerName)
 	if err == nil {
-		t.Fatal("GetBroker did not fail")
+		t.Fatal("Get did not fail")
 	}
 	if b != nil {
 		t.Fatalf("Got back a broker: %#v", b)
 	}
 }
 
 func TestAddBroker(t *testing.T) {
-	s := CreateMemStorage()
-	b := &model.ServiceBroker{GUID: "Test"}
-	cat := model.Catalog{
-		Services: []*model.Service{},
-	}
-	err := s.AddBroker(b, &cat)
+	bs := CreateMemStorage().Brokers()
+	b := &servicecatalog.Broker{ObjectMeta: kapi.ObjectMeta{Name: brokerName1}}
+	_, err := bs.Create(b)
 	if err != nil {
-		t.Fatalf("AddBroker failed with: %#v", err)
+		t.Fatalf("Create failed with: %s", err)
 	}
-	l, err := s.ListBrokers()
+	l, err := bs.List()
 	if len(l) != 1 {
 		t.Fatalf("Expected 1 broker, got %d", len(l))
 	}
-	b2, err := s.GetBroker("Test")
+	b2, err := bs.Get(brokerName1)
 	if err != nil {
-		t.Fatalf("GetBroker failed: %#v", err)
+		t.Fatalf("Get failed: %s", err)
 	}
 	if b2 == nil {
 		t.Fatal("Did not get back a broker")
 	}
-	if strings.Compare(b2.Name, b.Name) != 0 {
-		t.Fatalf("Names don't match, expected: '%s', got '%s'", b.Name, b2.Name)
+	if b2 != b {
```

Comment:
Strong opinion incoming...

So the general strategy behind this test case is that we're adding a broker to storage, then we're reading the same broker (or what we think is the same broker) back from storage as a means of asserting that addition worked. Comparing the values of a single field of the object you put in to the object you got back isn't a great heuristic. (Think how many times the broker name "Test1" is reused over the course of this test suite!) It's better to make a stronger assertion...

How can we? Well... here's the strong opinion part. It's acceptable for unit tests to know something about how the components under test are implemented. To defend this point of view, I've often invoked a mantra: "e2e tests verify you built the _right thing_, unit tests verify you built the _thing right_." If you buy into that, it's ok that these tests know that the object they get back from what is _known_ to be an in-memory implementation of storage should be the _exact_ object they put in.